Userland/Libraries/LibWeb/DOM/Document.cpp

```diff
@@ -1104,18 +1128,20 @@ void Document::update_layout()
         }
     }
 
-    Layout::LayoutState layout_state;
+    if (!m_just_typed_into_input_box) {
+        if (m_layout_state)
+            delete m_layout_state;
+        m_layout_state = new Layout::LayoutState();
```

Comment:
We try to avoid using raw pointers and new/delete. An `OwnPtr` would be better for this.

Reply:
![image](https://github.com/user-attachments/assets/187f8297-70a8-4158-83f1-7893aa29dd1e)
How do we construct or transfer values for an OwnPtr?  Specifically an OwnPtr<LayoutState> -- I get the following errors:
![image](https://github.com/user-attachments/assets/1db0dae8-97c1-46df-a0a1-d1a000f15ef6)
![image](https://github.com/user-attachments/assets/ca957ed4-f537-48d3-a48f-863a1e86ac97)

Reply:
Hmm, that second error might be because you're dereferencing the pointer. But the first should work. Does `#include <LibWeb/Layout/LayoutState.h>` help with that?